packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -198,6 +199,138 @@ class RenderEditable extends RenderBox {
 
   Rect _lastCaretRect;
 
+  bool _rightArrowDown = false;
+  bool _leftArrowDown = false;
+  bool _upArrowDown = false;
+  bool _downArrowDown = false;
+
+  static const int _kLeftArrowCode = 21;
+  static const int _kRightArrowCode = 22;
+  static const int _kUpArrowCode = 19;
+  static const int _kDownArrowCode = 20;
+
+  int _extentOffset = -1;
+  int _baseOffset = -1;
+
+  int _previousCursorLocation = -1;
+  bool _resetCursor = false;
+
+  void _keyEvent(RawKeyEvent keyEvent){
```

Comment:
Thank you for adding the documentation - it is a lot easier to follow what is going on here.